code/modules/mob/living/simple_animal/friendly/cat.dm

```diff
@@ -37,26 +37,99 @@
 	icon_resting = "cat_rest"
 	gender = FEMALE
 	gold_core_spawnable = CHEM_MOB_SPAWN_INVALID
+	var/list/family = list()
+	var/memory_saved = 0
+	var/list/children = list() //Actual mob instances of children
+	var/cats_deployed = 0
 
-/mob/living/simple_animal/pet/cat/handle_automated_action()
+/mob/living/simple_animal/pet/cat/Runtime/New()
+	Read_Memory()
 	..()
 
+/mob/living/simple_animal/pet/cat/Runtime/Life()
+	if(!cats_deployed && ticker.current_state >= GAME_STATE_SETTING_UP)
+		Deploy_The_Cats()
+	if(!stat && ticker.current_state == GAME_STATE_FINISHED && !memory_saved)
+		Write_Memory()
+	..()
+
+/mob/living/simple_animal/pet/cat/Runtime/make_babies()
+	var/mob/baby = ..()
+	if(baby)
+		children += baby
+		return baby
+
+/mob/living/simple_animal/pet/cat/Runtime/death()
+	if(!memory_saved)
+		Write_Memory(1)
+	..()
+
+/mob/living/simple_animal/pet/cat/Runtime/proc/Read_Memory()
+	var/savefile/S = new /savefile("data/npc_saves/Runtime.sav")
+	S["family"] 			>> family
+
+	if(isnull(family))
+		family = list()
+
+/mob/living/simple_animal/pet/cat/Runtime/proc/Write_Memory(dead)
+	var/savefile/S = new /savefile("data/npc_saves/Runtime.sav")
+	family = list()
+	if(!dead)
+		for(var/mob/living/simple_animal/pet/cat/kitten/C in children)
+			if(istype(C,type) || C.stat || !C.z || !C.butcher_results)
+				continue
+			if(C.type in family)
+				family[C.type] += 1
+			else
+				family[C.type] = 1
+	S["family"]				<< family
+	memory_saved = 1
+
+/mob/living/simple_animal/pet/cat/Runtime/proc/Deploy_The_Cats()
+	cats_deployed = 1
+	for(var/cat_type in family)
+		if(family[cat_type] > 0)
+			for(var/i in 1 to min(family[cat_type],100)) //Limits to about 500 cats, you wouldn't think this would be needed (BUT IT IS)
+				new cat_type(loc)
+
+
+/mob/living/simple_animal/pet/cat/Life()

```

Comment:
This should be:
```
/mob/living/simple_animal/pet/cat/handle_automated_action()
	..()
```
then exclude the `if(!stat && !buckled && !client)` and have the rest of the code execute.